### ISSUE TITLE
[qt] Duplicate CAN plugin on static build

### ISSUE THREAD

**Issue by jgeudens** [bincrafters/community#838]
### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **conan-qt**
  * Operating System+version: **Windows 10 64bit**
  * Compiler+version: **MinGW-5.3**

## conan comman ##
```
conan install .. -g virtualenv -s os=Windows -s arch=x86 -s compiler=gcc -s compiler.version=5.3 -s compiler.libcxx=libstdc++11 --build missing
```

### conanfile ###
```
[requires]
	qt/5.12.4@bincrafters/stable

[generators]
	qmake

[options]
	Qt:config=-static-runtime
	Qt:shared=False
	Qt:qtserialbus=True
	Qt:openssl=True
```

## Error message during using the toolchain ##
```
C:\Qt_Static_5.12.4\plugins\canbus\libqttinycanbus.a(tinycanbackend.o):-1: error: duplicate section `.rdata$_ZZZ14resolveSymbols...' has different size
C:\Qt_Static_5.12.4\plugins\canbus\libqtvectorcanbus.a(vectorcanbackend.o):-1: error: duplicate section `.rdata$_ZZZ14resolveSymbols...' has different size
```

## Report  ##
I generate a static compiled Qt toolchain with qserialbus. The qt compilation of the conan recipe succeeds without errors. But when I compile application with the customer static toolchain, I get the error message mentioned above. The compilation succeeds and the output is a complete statically linked application.

**Comment by ericLemanissier** [comment]
In order to target the cause of your problem more precisely, could you try to remove `Qt:config=-static-runtime`, and tell us if the error disappears ? I am not sure static libstdc++ is fully functional on mingw.

**Comment by jgeudens** [comment]
If you want, I can try it this evening, but I have added that statement when I encountered a previous error (https://github.com/bincrafters/community/issues/572). While building my application, the toolchain couldn't find several required libraries. I then added static-runtime to make sure MinGW was completly static linked. After that it worked.

The strange thing for this issue is that the toolchain works correctly. It can compile and link. Also running on a different computer without GCC and Qt works. It is just that it appears to be building two separate plugins for CAN for QT that conflicts with eachother.

**Comment by ericLemanissier** [comment]
An other idea: you should probably remove `[generators] qmake` from your conanfile. It is necessary only for libraries qmake does not know. qmake knows everything about the Qt version you built, so it does not need any information from conan to build your project.
On the other hand, you should probably add  `[generators] qt`, which will generate a qt.conf file in your build directory. You should move this file next to your executable before starting it.
To solve the problem you had in #572 without using `-static-runtime`, you have to make sure that
- either the path to missing dll  libstdc++-6.dll and libgcc_s_dw2-1.dll is present in the `PATH` environment variable when you execute your application
- or use [windeployqt ](https://doc.qt.io/qt-5/windows-deployment.html) (built with option qt:qttools=True) 

One last issue, the name of the recipe is all lowercase since few versions, so you should use only `qt`, not `Qt`

**Comment by jgeudens** [comment]
Thank you for your feedback. I make the changes to my recipe.

But I would like to distribute a full statically linked version of my application. Meaning 1 executable that contains everything required. So I would suppose that "-static-runtime" is still required then.

I'm starting to think that this issue of duplicate plugins is more related to building Qt from sources instead of specifically to the conan recipe.

**Comment by ericLemanissier** [comment]
This recipe is now in the Conan Center Index.

Please migrate to the recipe version there.
If this is still an issue, please create a new issue in the Conan Center Index issue tracker.

Thanks!

https://conan.io/center/qt
